rpc/geth.go

```diff
@@ -404,12 +424,30 @@ func (client *GethClient) GetERC20TokenMetadata(token []byte) (*types.ERC20Metad
 	g.Go(func() error {
 		decimals, err := contract.Decimals(nil)
 		if err != nil {
-			return fmt.Errorf("error retrieving decimals: %v", err)
+			return fmt.Errorf("error retrieving decimals: %w", err)
 		}
 		ret.Decimals = big.NewInt(int64(decimals)).Bytes()
 		return nil
 	})
+
+	g.Go(func() error {
+		rate, err := oracle.GetRateToEth(nil, common.BytesToAddress(token), false)
+		if err != nil {
+			return fmt.Errorf("error calling oneinchoracle.GetRateToEth: %w", err)
+		}
+		ethRate = rate
+		return nil
+	})
+
 	err = g.Wait()
+	if err != nil {
+		return ret, err
+	}
+
+	num := new(big.Int).Exp(big.NewInt(10), new(big.Int).SetBytes(ret.Decimals), nil)
+	den := new(big.Int).Exp(big.NewInt(10), big.NewInt(18), nil)
+	price := new(big.Int).Quo(new(big.Int).Mul(ethRate, num), den)
```

Comment:
Same question as for Erigon.
Is it intended here that you lose all after comma digits like for a normal int division?